sdk/eventgrid/Azure.Messaging.EventGrid/README.md

```diff
@@ -195,30 +218,6 @@ foreach (EventGridEvent egEvent in egEvents)
     }
 }
 ```
-Here is an example calling `GetData<T>()` using CloudEvents. In order to deserialize to the correct type, the `EventType` property (`Type` for CloudEvents) helps distinguish between different events.
-```csharp Snippet:DeserializePayloadUsingGenericGetData
-foreach (CloudEvent cloudEvent in cloudEvents)
-{
-    switch (cloudEvent.Type)
-    {
-        case "Contoso.Items.ItemReceived":
-            // By default, GetData uses JsonObjectSerializer to deserialize the payload
-            ContosoItemReceivedEventData itemReceived = cloudEvent.GetData<ContosoItemReceivedEventData>();
-            Console.WriteLine(itemReceived.ItemSku);
-            break;
-        case "MyApp.Models.CustomEventType":
-            // One can also specify a custom ObjectSerializer as needed to deserialize the payload correctly
-            TestPayload testPayload = await cloudEvent.GetDataAsync<TestPayload>(myCustomSerializer);
-            Console.WriteLine(testPayload.Name);
-            break;
-        case "Microsoft.Storage.BlobDeleted":
-            // Example for deserializing system events using GetData<T>
-            StorageBlobDeletedEventData blobDeleted = cloudEvent.GetData<StorageBlobDeletedEventData>();
-            Console.WriteLine(blobDeleted.BlobType);
-            break;
-    }
-}
-```
```

Comment:
> - An `InvalidCastException` will be thrown during `GetData<T>()` if the event data cannot be cast to the specified type.

I suppose this is technically true, but I think the more likely error would be a deserialization exception when attempting to deserialize. If using STJ, it would be a JsonException I think.